File: tests/post.py
#!/usr/bin/python

import os
import sys

# Get the content length from the environment variables
content_length = int(os.environ.get('CONTENT_LENGTH', 0))

# Read the POST data from stdin
post_data = sys.stdin.read(content_length)

# Now, post_data contains the request body data (e.g., form data or JSON)

print("\r\n")
# Print the HTML response
print("<html>")
print("<body>")
print("<h1 style=\"color:dodgerblue;\">Data Received (CGI TEST)</h1>")
print("<p>Hi: </p>")
print("<pre>%s</pre>" % post_data)
print("</body>")
print("</html>")

File: tests/post.php
#!/usr/bin/php

<?php

// Get the content length from the environment variables
$contentLength = (int) $_SERVER['CONTENT_LENGTH'];

// Read the POST data from stdin
$postData = fread(STDIN, $contentLength);

// Now, $postData contains the request body data (e.g., form data or JSON)

// Print the HTML response
echo "<html>\n";
echo "<body>\n";
echo "<h1 style=\"color: darkviolet;\">Data Received (CGI TEST)</h1>\n";
echo "<p>Hi: </p>\n";
echo "<pre>" . htmlspecialchars($postData) . "</pre>\n";
echo "</body>\n";
echo "</html>\n";

?>

